pkgs/tools/misc/pywb/default.nix

```diff
@@ -0,0 +1,207 @@
+          # We're patching this so it never fails.
+          "test_searches_for_c_stdlib_and_raises_if_missing"
+
+          "test_append"
+          "test_decr"
+        ];
+        # See https://github.com/NixOS/nixpkgs/issues/7307.
+        patchPhase = let
+          ext = stdenv.hostPlatform.extensions.sharedLibrary;
+        in
+          ''substituteInPlace fakeredis.py --replace \
+            "find_library('c')" "'${stdenv.cc.libc}/lib/libc${ext}.6'"
+        '';
+      });
+      redis = super.redis.overridePythonAttrs (old: rec {
+        version = "2.10.6";
+        src = old.src.override {
+          inherit version;
+          sha256 = "sha256-oiypk86ili27WI+fMNABWsSvzEW+4n05eMDb6el8bA8=";
+        };
+        pythonImportsCheck = [ ];
+      });
+      werkzeug = super.werkzeug.overridePythonAttrs (old: rec {
+        version = "1.0.1";
+        src = old.src.override {
+          inherit version;
+          sha256 = "6c80b1e5ad3665290ea39320b91e1be1e0d5f60652b964a3070216de83d2e47c";
+        };
+        checkInputs = old.checkInputs ++ (with self; [
+          requests
+        ]);
+        disabledTests = old.disabledTests ++ [
+          # ResourceWarning: unclosed file
+          "test_basic"
+          "test_date_to_unix"
+          "test_easteregg"
+          "test_file_rfc2231_filename_continuations"
+          "test_find_terminator"
+          "test_save_to_pathlib_dst"
+        ];
+        disabledTestPaths = old.disabledTestPaths ++ [
+          # ResourceWarning: unclosed file
+          "tests/test_http.py"
+          "tests/test_local.py"
+        ];
+      });
+    };
+  };
+in
+python.pkgs.buildPythonApplication rec {
+  name = "pywb";
+  version = "2.6.8";
+
+  src = fetchFromGitHub {
+    owner = "webrecorder";
+    repo = "pywb";
+    rev = "v-${version}";
+    sha256 = "sha256-+f0DLTjVZk/A3j02RRHzSgV1ozqUG3pg+cEqyTFc5ms=";
+    fetchSubmodules = true;
+  };
+
+  propagatedBuildInputs = with python.pkgs; [
+    brotlipy
+    chardet
+    fakeredis
+    gevent
+    jinja2
+    markupsafe
+    portalocker
+    py3amf
+    pytest
+    python-dateutil
+    pyyaml
+    redis
+    requests
+    six
+    surt
+    ua-parser
+    warcio
+    webassets
+    webencodings
+    werkzeug
+    wsgiprox
+  ];
+
+  checkInputs = with python.pkgs; [
+    pytest
+    mock
+    webtest
+    urllib3
+    httpbin
+    flask
+    ujson
+    lxml
+    fakeredis
+  ];
+
+  testExpression = optionalString (disabledTests != [])
+    "-k 'not ${concatStringsSep " and not " disabledTests}'";
+
+  checkPhase = ''
+    py.test tests/ ${testExpression}
+  '';
+
+  disabledTests = [
+    # 400 Bad Request.
+    "test_integration"
+    "test_live_rewriter"
+    "test_redirect_classic"
+    "test_socks"
+    "test_cert_req"
+    "test_force_https"
+  ];
+
+  meta = {
+    description = "Python web archiving toolkit for creating and replaying web archives";
+    homepage = https://github.com/webrecorder/pywb;
+    license = licenses.gpl3;
```

Comment:
Can you please clarify which gpl3? See https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/unclear-gpl.md

Reply:
It seems that pywb uses gplv3Plus (https://github.com/webrecorder/pywb) and wombat (also cloned since we're fetching submodules) uses agplv3Plus (https://github.com/webrecorder/wombat/blob/main/LICENSE.md). However, we're not building wombat for the time being but I've still specified both licenses.